config/utils.ts

```diff
@@ -0,0 +1,447 @@
+    throw new FileSystemError(
+      { cause: err },
+      {
+        filepath: configPath,
+        operation: 'write',
+      }
+    );
+  }
+}
+
+function getAccountType(sandboxAccountType?: string): AccountType {
+  if (sandboxAccountType) {
+    if (sandboxAccountType.toUpperCase() === 'DEVELOPER') {
+      return HUBSPOT_ACCOUNT_TYPES.DEVELOPMENT_SANDBOX;
+    }
+    if (sandboxAccountType.toUpperCase() === 'STANDARD') {
+      return HUBSPOT_ACCOUNT_TYPES.STANDARD_SANDBOX;
+    }
+  }
+  return HUBSPOT_ACCOUNT_TYPES.STANDARD;
+}
+
+export function normalizeParsedConfig(
+  parsedConfig: HubSpotConfig & DeprecatedHubSpotConfigFields
+): HubSpotConfig {
+  if (parsedConfig.portals) {
+    parsedConfig.accounts = parsedConfig.portals.map(account => {
+      if (account.portalId) {
+        account.accountId = account.portalId;
+        delete account.portalId;
+      }
+      return account;
+    });
+    delete parsedConfig.portals;
+  }
+
+  if (parsedConfig.defaultPortal) {
+    const defaultAccount = getConfigAccountByInferredIdentifier(
+      parsedConfig.accounts,
+      parsedConfig.defaultPortal
+    );
+
+    if (defaultAccount) {
+      parsedConfig.defaultAccount = defaultAccount.accountId;
+    }
+    delete parsedConfig.defaultPortal;
+  }
+
+  if (parsedConfig.defaultMode) {
+    parsedConfig.defaultCmsPublishMode = parsedConfig.defaultMode;
+    delete parsedConfig.defaultMode;
+  }
+
+  parsedConfig.accounts.forEach(account => {
+    if (!account.accountType) {
+      account.accountType = getAccountType(account.sandboxAccountType);
+    }
+  });
+
+  return parsedConfig;
+}
+
+export function parseConfig(configSource: string): HubSpotConfig {
+  let parsedYaml: HubSpotConfig & DeprecatedHubSpotConfigFields;
+
+  try {
+    parsedYaml = yaml.load(configSource) as HubSpotConfig &
+      DeprecatedHubSpotConfigFields;
+  } catch (err) {
+    throw new Error(i18n('config.utils.parseConfig.error'), { cause: err });
+  }
+
+  return normalizeParsedConfig(parsedYaml);
+}
+
+export function buildConfigFromEnvironment(): HubSpotConfig {
+  const apiKey = process.env[ENVIRONMENT_VARIABLES.HUBSPOT_API_KEY];
+  const clientId = process.env[ENVIRONMENT_VARIABLES.HUBSPOT_CLIENT_ID];
+  const clientSecret = process.env[ENVIRONMENT_VARIABLES.HUBSPOT_CLIENT_SECRET];
+  const personalAccessKey =
+    process.env[ENVIRONMENT_VARIABLES.HUBSPOT_PERSONAL_ACCESS_KEY];
+  const accountIdVar =
+    process.env[ENVIRONMENT_VARIABLES.HUBSPOT_ACCOUNT_ID] ||
+    process.env[ENVIRONMENT_VARIABLES.HUBSPOT_PORTAL_ID];
+  const refreshToken = process.env[ENVIRONMENT_VARIABLES.HUBSPOT_REFRESH_TOKEN];
+  const hubspotEnvironment =
+    process.env[ENVIRONMENT_VARIABLES.HUBSPOT_ENVIRONMENT];
+  const httpTimeoutVar = process.env[ENVIRONMENT_VARIABLES.HTTP_TIMEOUT];
+  const httpUseLocalhostVar =
+    process.env[ENVIRONMENT_VARIABLES.HTTP_USE_LOCALHOST];
+  const allowUsageTrackingVar =
+    process.env[ENVIRONMENT_VARIABLES.ALLOW_USAGE_TRACKING];
```

Comment:
We now support new env variables for `HTTP_TIMEOUT`, `ALLOW_USAGE_TRACKING` and other things that config files supported but environment configs didn't